java-checks/src/main/java/org/sonar/java/checks/ThrowCheckedExceptionCheck.java

```diff
@@ -41,18 +47,53 @@
 @SqaleConstantRemediation("1h")
 public class ThrowCheckedExceptionCheck extends SubscriptionBaseVisitor {
 
+  private Deque<MethodTree> methods = new LinkedList<>();
+
   @Override
   public List<Tree.Kind> nodesToVisit() {
-    return ImmutableList.of(Tree.Kind.THROW_STATEMENT);
+    return ImmutableList.of(Tree.Kind.METHOD, Tree.Kind.THROW_STATEMENT);
   }
 
   @Override
   public void visitNode(Tree tree) {
-    ThrowStatementTree throwStatementTree = (ThrowStatementTree) tree;
-    Type symbolType = throwStatementTree.expression().symbolType();
-    if (symbolType.isSubtypeOf("java.lang.Exception") && !symbolType.isSubtypeOf("java.lang.RuntimeException")) {
-      reportIssue(throwStatementTree.expression(), "Remove the usage of the checked exception '" + symbolType.name() + "'.");
+    if (tree.is(Tree.Kind.METHOD)) {
+      methods.push((MethodTree) tree);
+    } else {
+      ThrowStatementTree throwStatementTree = (ThrowStatementTree) tree;
+      Type symbolType = throwStatementTree.expression().symbolType();
+      if (symbolType.isSubtypeOf("java.lang.Exception") && !symbolType.isSubtypeOf("java.lang.RuntimeException") && !isFromMethodOverride(symbolType, methods)) {
+        reportIssue(throwStatementTree.expression(), "Remove the usage of the checked exception '" + symbolType.name() + "'.");
+      }
+    }
+  }
+
+  @Override
+  public void leaveNode(Tree tree) {
+    if (tree.is(Tree.Kind.METHOD)) {
+      methods.pop();
+    }
+  }
+
+  private static boolean isFromMethodOverride(Type exceptionType, Deque<MethodTree> methods) {
```

Comment:
If you don't make this static, no need for the second parameter as methods is a field of your class.